Reduce use of `static_cast<>()` in WebCore

cdumez opened the pull request:
#### 5436bdc1a6f89ad71e525a4920956d70be10f939
<pre>
Reduce use of `static_cast&lt;&gt;()` in WebCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=285176">https://bugs.webkit.org/show_bug.cgi?id=285176</a>

Reviewed by Geoffrey Garen and Darin Adler.

* Source/WebCore/Modules/mediasource/MediaSource.h:
(WebCore::MediaSource::lookup):
(isType):
* Source/WebCore/Modules/mediasource/MediaSourceRegistry.cpp:
(WebCore::MediaSourceRegistry::registerURL):
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::CompositeEditCommand::applyCommandToComposite):
* Source/WebCore/editing/CompositeEditCommand.h:
(isType):
* Source/WebCore/editing/EditCommand.cpp:
(WebCore::compositionIfPossible):
* Source/WebCore/editing/EditCommand.h:
(isType):
(WebCore::toSimpleEditCommand): Deleted.
* Source/WebCore/fileapi/Blob.cpp:
(WebCore::BlobURLRegistry::registerURL):
* Source/WebCore/fileapi/Blob.h:
(isType):
* Source/WebCore/html/URLRegistry.h:
* Source/WebCore/inspector/DOMPatchSupport.cpp:
(WebCore::DOMPatchSupport::patchDocument):

Canonical link: <a href="https://commits.webkit.org/288311@main">https://commits.webkit.org/288311@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7ffce233958398b187efaef2715ed46d4be81be6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/82419 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/2083 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/36509 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/87551 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/33480 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/84524 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/2154 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/9970 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/64240 "Found 1 new test failure: imported/w3c/web-platform-tests/html/semantics/selectors/case-sensitivity/values.window.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/21995 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/85488 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/1532 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/74994 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/44517 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/1431 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/29179 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/32521 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/72723 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/29814 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/88907 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/9725 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/6988 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/72641 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/9951 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/70808 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/71858 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/15996 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/15006 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/1098 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12797 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/9678 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/15199 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/9552 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/13018 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/11322 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/Modules/mediasource/MediaSourceRegistry.cpp:
```diff
@@ -57,9 +57,9 @@ void MediaSourceRegistry::registerURL(const ScriptExecutionContext& context, con
-    MediaSource& source = static_cast<MediaSource&>(registrable);
-    source.addedToRegistry();
-    m_mediaSources.add(urlString, std::pair { RefPtr { &source }, context.identifier() });
+    Ref source = downcast<MediaSource>(registrable);
```
Can we do `WTFMove(registrable)` here?

cdumez replied:
`registrable` is a `URLRegistrable&` so `WTFMove()` wouldn't be beneficial here unless I'm missing something.

cdumez committed:
Reduce use of `static_cast<>()` in WebCore
https://bugs.webkit.org/show_bug.cgi?id=285176

Reviewed by Geoffrey Garen and Darin Adler.

* Source/WebCore/Modules/mediasource/MediaSource.h:
(WebCore::MediaSource::lookup):
(isType):
* Source/WebCore/Modules/mediasource/MediaSourceRegistry.cpp:
(WebCore::MediaSourceRegistry::registerURL):
* Source/WebCore/editing/CompositeEditCommand.cpp:
(WebCore::CompositeEditCommand::applyCommandToComposite):
* Source/WebCore/editing/CompositeEditCommand.h:
(isType):
* Source/WebCore/editing/EditCommand.cpp:
(WebCore::compositionIfPossible):
* Source/WebCore/editing/EditCommand.h:
(isType):
(WebCore::toSimpleEditCommand): Deleted.
* Source/WebCore/fileapi/Blob.cpp:
(WebCore::BlobURLRegistry::registerURL):
* Source/WebCore/fileapi/Blob.h:
(isType):
* Source/WebCore/html/URLRegistry.h:
* Source/WebCore/inspector/DOMPatchSupport.cpp:
(WebCore::DOMPatchSupport::patchDocument):

Canonical link: https://commits.webkit.org/288311@main
